sql-plugin/src/main/scala/org/apache/spark/sql/rapids/RapidsCachingReader.scala

```diff
@@ -142,7 +142,7 @@ class RapidsCachingReader[K, C](
       try {
         val cachedIt = cachedBufferIds.iterator.map(bufferId => {
           GpuSemaphore.acquireIfNecessary(context)
-          val cb = withResource(catalog.acquireBuffer(bufferId)) { buffer =>
+          val cb = withResource(catalog.acquireBuffer(bufferId, unspill = true)) { buffer =>
```

Comment:
should this iterator return a set of batches that are "to be" unspilled later. Giving the coalesce batches that is pulling from it a chance to batch the unspills in a coherent way?

Reply:
Yes, batched spilling can be a further optimization.